onnxruntime/core/graph/graph.cc

```diff
@@ -3652,8 +3653,10 @@ Status Graph::ReplaceInitializedTensorImpl(ONNX_NAMESPACE::TensorProto new_initi
 
   // New initializers data generally are within OrtValues
   // Small initializers are still stored inside TensorProto
-  ORT_RETURN_IF_NOT(utils::HasExternalDataInMemory(new_initializer) || !ort_value.IsAllocated(),
-                    "All TensorProtos are expected to point to an OrtValue");
+  ORT_RETURN_IF_NOT(utils::HasExternalDataInMemory(new_initializer) ||
+                        (!ort_value.IsAllocated() && !utils::HasExternalData(new_initializer)),
```

Comment:
This explanation isn't quite clear to me but maybe it's all the negations in the condition between the IF_NOT and the `!`s.

Reply:
updated. hopefully it's a bit clearer.